follower:
  controller_manager:
    ros__parameters:
      update_rate: 500

      follower_mpc:
        type: formation_mpc/FollowerMPC

      follower_joint_bcast:
        type: joint_state_broadcaster/JointStateBroadcaster

  follower_joint_bcast:
    ros__parameters:
      use_local_topics: True
      joints:
        - "follower/joint_x"
        - "follower/joint_y"
        - "follower/joint_rz"
        - "follower/ur/shoulder_pan_joint"
        - "follower/ur/shoulder_lift_joint"
        - "follower/ur/elbow_joint"
        - "follower/ur/wrist_1_joint"
        - "follower/ur/wrist_2_joint"
        - "follower/ur/wrist_3_joint"

  follower_mpc:
    ros__parameters:
      control_horizon_in_s: 1.0
      number_of_points: 5
      dt: 0.01
      cartesian_axis: [true, true, true, true, true, true]
      clik:
        active: true
        gain: 1.0
      pid:
        P: [1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0]
  #    ns: "follower"
      mobile_base_link: "follower/base_link"
      manipulator_base_link: "follower/ur/base_link"
      tool_frame: "follower/ur/tool0"
      map_frame: "map"
      mount_frame: "follower/mount_frame"
      payload_frame: "object"
      grasp_frame: "follower/grasp_point"
      joints:
        base:
          - "follower/joint_x"
          - "follower/joint_y"
          - "follower/joint_rz"
        arm:
          - "follower/ur/shoulder_pan_joint"
          - "follower/ur/shoulder_lift_joint"
          - "follower/ur/elbow_joint"
          - "follower/ur/wrist_1_joint"
          - "follower/ur/wrist_2_joint"
          - "follower/ur/wrist_3_joint"
      initial_positions:
        - 0.0
        - 0.0
        - 0.0
        - 0.0
        - -2.356
        - 1.57
        - 0.0
        - 0.0
        - 0.0
      command_interface:
  #      - "velocity"
        - "position"
      state_interface:
        - "position"
        - "velocity"
      cmd_vel_topic: "follower/cmd_vel"
      leader_topic: "leader/cmd_vel"
      cartesian_task_weight:
        active: true
        weight: 1.0
      scaling: 1.0
      base_max_vel: [5.0, 5.0, 5.0, 5.0, 5.0, 5.0]
      fake_base:
        package: "cooperative_transport_cell"
        path: "urdf/fake_base.urdf"
